### PR TITLE
adding 'only' option to the tests.py

### PR DESCRIPTION
**Pull request by ipdemes**

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -297,9 +308,18 @@ def run_tests(
     options=[],
     interop_tests=False,
     workers=None,
+    only_pattern=None,
 ):
     if interop_tests:
-        legate_tests.extend(glob.glob("tests/interop/*.py"))
+        legate_tests.extend(glob.glob("tests/interop/*.py"))  # noqa: F823
+
+    if only_pattern is not None:
```
Comment (magnatelee) [inline test.py:316]: Let's not repeat this, unless there's a reason good reason to do so. Can't you factor this and the same code above out?

File: test.py [hunk 2]
```diff
@@ -297,9 +308,18 @@ def run_tests(
     options=[],
     interop_tests=False,
     workers=None,
+    only_pattern=None,
 ):
     if interop_tests:
-        legate_tests.extend(glob.glob("tests/interop/*.py"))
+        legate_tests.extend(glob.glob("tests/interop/*.py"))  # noqa: F823
```
Comment (magnatelee) [inline test.py:314]: Why do you need to override F823? We generally don't want overrides like this, unless there's no way around it. (as far as I know we use this override only for very long function names.)

Comment (magnatelee) [inline test.py:314]: Why do you need to override F823? We generally don't want overrides like this, unless there's no way around it. (as far as I know we use this override only for very long function names.)

Comment (ipdemes) [reply test.py]: it was a leftover, removed

File: test.py (hunk 1 above)
Comment (ipdemes) [reply test.py:318]: Sure, done

**Comment by ipdemes** [comment]
@magnatelee : can I merge this?

**Comment by magnatelee** [comment]
Yes. Please merge it when you can.

**Pull request merged**